.github/workflows/database.yaml

```diff
@@ -90,40 +129,18 @@ jobs:
           PGPORT: 5432
 
       - name: Upload sanitised backup
+        if: github.event_name == 'schedule'
         uses: actions/upload-artifact@v4
         with:
           name: backup-sanitised
           path: backup-sanitised.sql.gz
           retention-days: 3
 
-      - id: key-vault-name
-        if: failure()
-        shell: bash
-        run: echo "value=$(make -s production print-infrastructure-key-vault-name)" >> $GITHUB_OUTPUT
-
-      - uses: Azure/get-keyvault-secrets@v1
-        if: failure()
-        id: key-vault-secrets
-        with:
-          keyvault: ${{ steps.key-vault-name.outputs.value }}
-          secrets: "SLACK-WEBHOOK"
-
-      - name: Notify Slack channel on job failure
-        if: failure()
-        uses: rtCamp/action-slack-notify@v2
-        env:
-          SLACK_USERNAME: CI Deployment
-          SLACK_TITLE: Database backup failure
-          SLACK_MESSAGE: Production database backup job failed
-          SLACK_WEBHOOK: ${{ steps.key-vault-secrets.outputs.SLACK-WEBHOOK }}
-          SLACK_COLOR: failure
-          SLACK_FOOTER: Sent from backup-production job in database-backups workflow
-
   restore-preproduction:
     name: Restore preproduction
-    needs: [backup-production]
+    needs: [backup]
+    if: ${{ inputs.environment == 'production' || github.event_name == 'schedule' }}
```

Comment:
wouldn't this have to be just `github.event_name == 'schedule'` like the others above?